mysite/lti/views.py

```diff
@@ -101,21 +109,65 @@ def lti_redirect(request, course_id=None, unit_id=None):
 
     if not user.is_linked:
         user.create_links()
-
     user.login(request)
+
+    if course_ref:
+        course_id = course_ref.course.id
+    elif Role.INSTRUCTOR in roles:
+        return redirect(reverse('lti:create_courseref'))
+    else:
+        return redirect(reverse('ct:home'))
+
     user.enroll(roles, course_id)
 
     if user.is_enrolled(roles, course_id):
         # Redirect to course or unit page considering users role
         if not unit_id:
             dispatch = 'ct:course_student'
-            if roles == 'prof':
+            if roles == Role.INSTRUCTOR:
```

Comment:
Actually

B.3    Roles
Some of the commonly used roles from LIS include Learner, Instructor, Administrator, TeachingAssistant, ContentDeveloper, and Mentor.  Multiple roles can be included, separated by commas.  TC systems should include as many roles as appropriate for the user (i.e., more roles are better).  TC systems should be aware that simple TPs will key off the presence or absence of the Instructor role and group users into those with the Instructor role (read-write-configure) and those without the Instructor role (read).

http://www.imsglobal.org/lti/blti/bltiv1p0/ltiBLTIimgv1p0.html#_Toc261271991

So I think we need to fix and parse it